lmdeploy/model.py

```diff
@@ -440,242 +431,135 @@ def __init__(self,
-        self.temperature = temperature
-
-        self.im_start = im_start
-        self.im_end = im_end
-        self.system = system
-        self.stop_words = stop_words
+        super().__init__(system=system,
+                         meta_instruction=meta_instruction,
+                         eosys=eosys,
+                         user=user,
+                         eoh=eoh,
+                         assistant=assistant,
+                         eoa=eoa,
+                         separator=separator,
+                         stop_words=stop_words,
+                         session_len=session_len,
+                         **kwargs)
 
     @classmethod
-    def from_config(cls, **kwargs):
-        """Build model from ModelConfig parameters."""
-        # TODO system, eosys, user, eoh, assistant, eoa.
-        mapping = {'meta_instruction': 'system'}
-        new_kwargs = {
-            mapping.get(key, key): value
-            for key, value in kwargs.items()
-        }
-        return cls(**new_kwargs)
-
-    def decorate_prompt(self, prompt, sequence_start=True):
-        assert self.capability == 'chat', \
-            f'{type(self).__name__} has no capability of {self.capability}'
-        if sequence_start:
-            return f'{self.im_start}system\n{self.system}{self.im_end}' \
-                   f'\n{self.im_start}user\n{prompt}{self.im_end}' \
-                   f'\n{self.im_start}assistant\n'
-
-        return f'\n{self.im_start}user\n{prompt}{self.im_end}' \
-               f'\n{self.im_start}assistant\n'
-
-    def messages2prompt(self, messages, sequence_start=True):
-        """Return the prompt that is concatenated with other elements in the
-        chat template.
+    def match(cls, model_path: str) -> Optional[str]:
+        """Return the model_name that was registered to MODELS.
 
         Args:
-            messages (str | List): user's input prompt
-        Returns:
-            str: the concatenated prompt
+            model_path (str): the model path used for matching.
         """
-        if isinstance(messages, str):
-            return self.get_prompt(messages, sequence_start)
-        system, users, assistants = self._translate_messages(messages)
-        system = self.system if not system else system
-        ret = f'{self.im_start}system\n{system}{self.im_end}'
-        for user, assistant in zip(users, assistants):
-            if assistant:
-                ret += f'\n{self.im_start}user\n{user}{self.im_end}' \
-                       f'\n{self.im_start}assistant\n{assistant}'
-            else:
-                ret += f'\n{self.im_start}user\n{user}{self.im_end}' \
-                       f'\n{self.im_start}assistant\n'
-        return ret
+        if 'qwen' in model_path.lower():
+            return 'qwen-chat'
 
 
 @MODELS.register_module(name='codellama')
 class CodeLlama(Llama2):
 
     def __init__(self,
-                 system='',
+                 meta_instruction='',
```

Comment:
it seems `session_len`, `suffix_first` and `stop_words` are unused ...

Reply:
Because `codellama` can do completion, infilling, and python tasks apart from chat.